uwblueprint/sistema
# Google OAuth Setup

# Notion ticket link
[Configure Google Auth and Simple Login](https://www.notion.so/uwblueprintexecs/Configure-Google-Auth-and-Simple-Login-0cd9d7acb1be44939d0cf584a1173473?pvs=4)


## Implementation description
I'm assuming the reader is familiar with how OAuth works. If they're not I'd recommend going through [this video](https://www.youtube.com/watch?v=t18YB3xDfXI).
auth.js reads the 3 environment variables from the `.env` files: 
- Chinemerem set up the Google Cloud credentials for OAuth and added them to Vault.
- The [Auth.js configuration](https://authjs.dev/getting-started/installation#setup-environment) requires an env variable called `AUTH_SECRET` to be set up. Did that locally and added the var to Vault.
- Followed this guide to access the `session` object: https://authjs.dev/getting-started/session-management/login
- Followed Auth.js's [Github OAuth tutorial](https://authjs.dev/guides/configuring-github) then made the modifications for converting that to a valid [Google OAuth](https://authjs.dev/getting-started/authentication/oauth) flow.


## Steps to test

1. Please populate environment variables from Vault as described in [this section](https://github.com/uwblueprint/sistema?tab=readme-ov-file#secrets) of the readme.
2. Please run `npm run dev`
3. Got to `http://localhost:3000`. Click sign in. You should be able to login with Google. 
4. Go back to `http://localhost:3000` to check the data we get from Google. 
5. Click on signout and confirm that you can see `unauthenticated`.


## What should reviewers focus on?

- Are the changes to `package.json` alright? Should they not be part of this PR since they are not strictly required for the ticket?
- Made some changes in tsconfig.json that I'd like to discuss verbally. The changes I've made have implications for configuring build tools & bundlers in the future. Please read [this page](https://www.typescriptlang.org/docs/handbook/modules/reference.html#paths-does-not-affect-emit) of the TypeScript docs on `paths` only informing TypeScript about the module aliases.

## Checklist
